<!DOCTYPE html>
<html lang="en"
>
<head>
    <title>iMetAMOS install - Research Notebook</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">



<link rel="canonical" href="http://nate-d-olson.github.io/lab_notebok/iMetAMOS install.html">

        <meta name="author" content="Nate Olson" />
        <meta name="keywords" content="iMetAMOS,Docker" />
        <meta name="description" content="Installing iMetAMOS, with an attempt to use docker. Unable to run with automated install. Ended up installing it locally on desktop with Ubuntu 14.04." />

        <meta property="og:site_name" content="Research Notebook" />
        <meta property="og:type" content="article"/>
        <meta property="og:title" content="iMetAMOS install"/>
        <meta property="og:url" content="http://nate-d-olson.github.io/lab_notebok/iMetAMOS install.html"/>
        <meta property="og:description" content="Installing iMetAMOS, with an attempt to use docker. Unable to run with automated install. Ended up installing it locally on desktop with Ubuntu 14.04."/>
        <meta property="article:published_time" content="2014-11-20" />
            <meta property="article:section" content="Micro RM" />
            <meta property="article:tag" content="iMetAMOS" />
            <meta property="article:tag" content="Docker" />
            <meta property="article:author" content="Nate Olson" />


    <!-- Bootstrap -->
        <link rel="stylesheet" href="http://nate-d-olson.github.io/lab_notebok/theme/css/bootstrap.min.css" type="text/css"/>
    <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/pygments/native.css" rel="stylesheet">
        <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/typogrify.css" rel="stylesheet">
    <link rel="stylesheet" href="http://nate-d-olson.github.io/lab_notebok/theme/css/style.css" type="text/css"/>

        <link href="http://nate-d-olson.github.io/lab_notebok/feeds/all.atom.xml" type="application/atom+xml" rel="alternate"
              title="Research Notebook ATOM Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
	<div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="http://nate-d-olson.github.io/lab_notebok/" class="navbar-brand">
Research Notebook            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                        <li >
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/ipynb.html">Ipynb</a>
                        </li>
                        <li class="active">
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/micro-rm.html">Micro rm</a>
                        </li>
                        <li >
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/productivity.html">Productivity</a>
                        </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><a href="http://nate-d-olson.github.io/lab_notebok/archives.html"><i class="fa fa-th-list"></i><span class="icon-label">Archives</span></a></li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->
<!-- Banner -->
<!-- End Banner -->
<div class="container">
    <div class="row">
        <div class="col-sm-9">

    <section id="content">
        <article>
            <header class="page-header">
                <h1>
                    <a href="http://nate-d-olson.github.io/lab_notebok/iMetAMOS install.html"
                       rel="bookmark"
                       title="Permalink to iMetAMOS install">
                        iMetAMOS&nbsp;install
                    </a>
                </h1>
            </header>
            <div class="entry-content">
                <div class="panel">
                    <div class="panel-body">
<footer class="post-info">
    <span class="label label-default">Date</span>
    <span class="published">
        <i class="fa fa-calendar"></i><time datetime="2014-11-20T00:00:00+01:00"> Thu 20 November 2014</time>
    </span>



<span class="label label-default">Tags</span>
	<a href="http://nate-d-olson.github.io/lab_notebok/tag/imetamos.html">iMetAMOS</a>
        /
	<a href="http://nate-d-olson.github.io/lab_notebok/tag/docker.html">Docker</a>
    
</footer><!-- /.post-info -->                    </div>
                </div>
                <h1>setting up&nbsp;iMetAMOS</h1>
<p>install in utilities directory but running in&nbsp;docker</p>
<p>wget https://github.com/marbl/metAMOS/archive/v1.5rc3.zip
unzip&nbsp;v1.5rc3.zip</p>
<p>key error for os.environ[&#8220;<span class="caps">SHELL</span>&#8221;] - install in utilities from&nbsp;desktop</p>
<p>Messages from running python <span class="caps">INSTALL</span>.py imetAMOS
To make things easier for you, you may want to copy/symlink the following
files into a directory in your <span class="caps">PATH</span>:
  /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken
  /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken-build
  /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken-filter
  /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken-mpa-report
  /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/kraken/bin/kraken-report
Could not import pysam, disabling.
Could not import psutil,&nbsp;disabling.</p>
<p>Kraken database download
tar (child): minikraken.tgz: Cannot open: Timer expired
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
mv: cannot stat ‘minikraken_*’: No such file or&nbsp;directory</p>
<p>KronaTools install error
syntax error at ./install.pl line 58, near &#8220;)&#8221;
  (Might be a runaway multi-line () string starting on line 37)
syntax error at ./install.pl line 80, near &#8220;}&#8221;
Illegal declaration of subroutine main::createDir at ./install.pl line&nbsp;89.</p>
<p>SignalP is not installed and is required for Prokka&#8217;s gram option. Please download it and add it to your path.
had to download from <span class="caps">DTU</span> website using <span class="caps">UMD</span> email&nbsp;address</p>
<p>no error when rerunning the install&nbsp;script</p>
<p>error with Ray not <span class="caps">MPI</span></p>
<p>Issues with minikraken database - tried downloading multiple times and still received an error regarding opening the&nbsp;database.</p>
<p>Running tests
run_pipeline_test.sh - issue with Krona tools
run_test2.sh issue with nb_classify
run_test_fast.sh - error cannot locate KronaTools.pm
run_test_amphora.sh - error with KronaTools when calling&nbsp;ImportPhylosift </p>
<p>Error with syntax error: bad fd number 
/bin/sh link to dash changed to bash following recomendations on 
http://stackoverflow.com/questions/15809060/sh-syntax-error-bad-fd-number
<code>sudo mv /bin/sh /bin/sh.orig
sudo ln -s /bin/bash /bin/sh</code></p>
<p>issue with KronaTools install - manually added links to perl scripts KronaTools directory between scirpts and&nbsp;bin</p>
<p>for error with fcb_models
downloaded from metamos cbcb ftp site extracted on laptop and used rsync to transfer to desktop into metaAMOS utilities/models&nbsp;directory</p>
<p>found potential bug in the KronaTools install&nbsp;script</p>
<p>for Ray install added <span class="caps">MPI</span> to path 
<em>   <code>sudo apt-get install libopenmpi-dev</code>
</em>   reran pyhton <span class="caps">INSTALL</span>.py imetAMOS 
*   able to install Ray!
after fixing bug with krona perl script able to run&nbsp;run_pipeline_test.sh</p>
<p>FRCbam install error related to boost
apt-get install libboost-system-dev libboost-isostreams-dev
still not able to&nbsp;install</p>
<p>installing <span class="caps">CA</span> - gmake not 
running&nbsp;test_ima.sh </p>
<h3>imetAMOS&nbsp;run</h3>
<p>error with <span class="caps">SRR1555300</span></p>
<div class="highlight"><pre>*****************************************************************
*************************ERROR***********************************
During findorfs, the following command failed with return code 2:
&gt;&gt; /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/prokka  --outdir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structurm: cannot remove ‘/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Logs/findorfs.ok’: No such file or directory
Traceback (most recent call last):
  File &quot;./runPipeline&quot;, line 984, in &lt;module&gt;
    verbose = 1)
  File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 2965, in pipeline_run
    raise job_errors
RethrownJobError: 



    Exception #1
      &#39;exceptions.TypeError(exceptions must be old-style classes or derived from BaseException, not NoneType)&#39; raised in ...
       Task = def validate.Validate(...):
       Job  = [[abyss.91.faa, idba-ud.91.faa, masurca.91.faa, sga.91.faa, soapdenovo2.91.faa, spades.91.faa, velvet.91.faa] -&gt; [validate.ok]]

    Traceback (most recent call last):
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 625, in run_pooled_job_without_exceptions
        return_value =  job_wrapper(param, user_defined_work_func, register_cleanup, touch_files_only)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 491, in job_wrapper_io_files
        ret_val = user_defined_work_func(*param)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/validate.py&quot;, line 690, in Validate
        FindORFS(&quot;%s/Assemble/out/%s.contig.cvg&quot;%(_settings.rundir, _settings.PREFIX), &quot;%s/Assemble/out/%s.faa&quot;%(_settings.rundir, _settings.PREFIX))
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/findorfs.py&quot;, line 380, in FindORFS
        findFastaORFs(_orf, &quot;%s/FindORFS/in/%s.asm.contig&quot;%(_settings.rundir, _settings.PREFIX), &quot;%s.ctg.fna&quot;%(_settings.PREFIX), &quot;%s.ctg.faa&quot;%(_settings.PREFIX), &quot;%s.ctg.gene.cvg&quot;%(_settings.PREFIX), &quot;%s.ctg.gene.map&quot;%(_settings.PREFIX), _min_ctg_len, _min_ctg_cvg)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/findorfs.py&quot;, line 337, in findFastaORFs
        run_process(_settings, &quot;%s/prokka %s --outdir %s/FindORFS/out/%s.prokka --prefix %s --force %s&quot;%(_settings.PROKKA,&quot;--fast&quot; if _run_fast else &quot;&quot;,  _settings.rundir, _settings.PREFIX, _settings.PREFIX, contigs), &quot;FindORFS&quot;)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/utils.py&quot;, line 1145, in run_process
        raise
    TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType


re/SRR155300/FindORFS/out/proba.prokka --prefix proba --force /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/in/proba.asm.contig

*************************DETAILS***********************************
Last 10 commands run before the error (/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Logs/COMMANDS.log)
|2014-11-25 05:44:09| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/.asm.tigr /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.asm.tigr
|2014-11-25 05:44:09| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.contig.cnt
|2014-11-25 05:44:09| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/.contig.cnt /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.contig.cnt
|2014-11-25 05:44:09| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.contig.cvg
|2014-11-25 05:44:09| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/.contig.cvg /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.contig.cvg
|2014-11-25 05:44:09| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.afg
|2014-11-25 05:44:09| rm /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Logs/findorfs.ok
|2014-11-25 05:44:09| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/in/proba.asm.contig
|2014-11-25 05:44:09| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Assemble/out/proba.asm.contig /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/in/proba.asm.contig
|2014-11-25 05:44:11| /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/prokka  --outdir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/out/proba.prokka --prefix proba --force /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/in/proba.asm.contig

Last 10 lines of output (/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/Logs/FINDORFS.log)
[05:44:10] Determined parallel version is 20130822
[05:44:10] Looking for &#39;prodigal&#39; - found /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/../binaries/linux/prodigal
[05:44:10] Determined prodigal version is 2.60
[05:44:10] Looking for &#39;sed&#39; - found /bin/sed
[05:44:10] Looking for &#39;signalp&#39; - found /usr/local/bin/signalp
[05:44:11] Determined signalp version is 4.1
[05:44:11] Looking for &#39;tbl2asn&#39; - found /usr/local/bin/tbl2asn
[05:44:11] Determined tbl2asn version is 23.9
[05:44:11] Using genetic code table 11.
[05:44:11] &#39;/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR155300/FindORFS/in/proba.asm.contig&#39; is not a readable non-empty FASTA file

Please veryify input data and restart MetAMOS. If the problem persists please contact the MetAMOS development team.
*************************ERROR***********************************
*****************************************************************[0m
Oops, MetAMOS finished with errors! see text in red above for details.

## Error with SRR1555298 run 
*** metAMOS running command: head -n 4 /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555298/Preprocess/out/lib1.fasta &gt; /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_struTraceback (most recent call last):
  File &quot;./runPipeline&quot;, line 984, in &lt;module&gt;
    verbose = 1)
  File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 2965, in pipeline_run
    raise job_errors
RethrownJobError: 



    Exception #1
      &#39;exceptions.ValueError(invalid literal for int() with base 10: &#39;42\n42&#39;)&#39; raised in ...
       Task = def assemble.Assemble(...):
       Job  = [idba-ud.67.run -&gt; idba-ud.67.asm.contig]

    Traceback (most recent call last):
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 625, in run_pooled_job_without_exceptions
        return_value =  job_wrapper(param, user_defined_work_func, register_cleanup, touch_files_only)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 491, in job_wrapper_io_files
        ret_val = user_defined_work_func(*param)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/assemble.py&quot;, line 537, in Assemble
        generic.execute(STEP_NAMES.ASSEMBLE, asmName.lower(), _settings)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 493, in execute
        return dispatch.execute()
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 342, in execute
        (havePE, haveMP, offset, input, pelist, mplist, techParams) = self.getLibInput()
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 251, in getLibInput
        if (int(libReadLen) &lt; self.readLen):
    ValueError: invalid literal for int() with base 10: &#39;42\n42&#39;


cture/SRR1555298/Preprocess/out/tmp.fasta

*** metAMOS running command: rm -f /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555298/Preprocess/out/tmp.fasta

Oops, MetAMOS finished with errors! see text in red above for details.
</pre></div>


<h2>Restarted <span class="caps">SRR1555300</span> run from&nbsp;terminal</h2>
<p>11/25/2014 </p>
<p>Trying to Run using the binary
nolson@Selenium:/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS_v1.5rc2_binary$ ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555298_1.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555298_2.fastq  -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555298</span>-b -i 500:700 -W iMetAMOS
Error: cannot find <span class="caps">BLAST</span> <span class="caps">DB</span> directory, yet path set via $<span class="caps">BLASTDB</span>: /var/lib/blastdb. Disabling blastdb dependent programs
Project dir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/<span class="caps">SRR1555298</span>-b successfully created!
Use runPipeline.py to start Pipeline
nolson@Selenium:/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS_v1.5rc2_binary$ ./runPipeline -v -d ../../analysis/bioinf/
genome_purity/        genome_structure/     get_fastq/            sequence_homogeneity/ sequence_purity/    <br />
nolson@Selenium:/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS_v1.5rc2_binary$ ./runPipeline -v -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555298</span>-b/
Warning: Cannot determine <span class="caps">OS</span>, defaulting to Linux
Warning: Cannot determine <span class="caps">OS</span> version, defaulting to 0.0
Warning: Cannot determine system type, defaulting to x86_64
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.11' not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.14&#8217; not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.8' not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.15&#8217; not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.14' not found (required by /lib/x86_64-linux-gnu/libtinfo.so.5)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.15&#8217; not found (required by /lib/x86_64-linux-gnu/libtinfo.so.5)
Error: cannot find <span class="caps">BLAST</span> <span class="caps">DB</span> directory, yet path set via $<span class="caps">BLASTDB</span>: /var/lib/blastdb. Disabling blastdb dependent programs
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.11' not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.14&#8217; not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.8' not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.15&#8217; not found (required by sh)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version <code>GLIBC_2.14' not found (required by /lib/x86_64-linux-gnu/libtinfo.so.5)
sh: /tmp/_MEIutl4pm/lib/libc.so.6: version</code>GLIBC_2.15&#8217; not found (required by /lib/x86_64-linux-gnu/libtinfo.so.5)
IOError: [Errno 2] No such file or directory:&nbsp;&#8216;/tmp/_MEIutl4pm/phylosift/phylosiftrc.orig&#8217;</p>
<h2>restarted <span class="caps">SRR155298</span>&nbsp;run</h2>
<p>./runPipeline for <span class="caps">SRR1555298</span> with -f assembly got the following&nbsp;error</p>
<div class="highlight"><pre>*** metAMOS running command: /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/prokka  --outdir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/out/proba.prokka --prefix proba --force /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig

[*****************************************************************
*************************ERROR***********************************
During findorfs, the following command failed with return code 2:
&gt;&gt; /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/prokka  --outdir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/out/proba.prokka --prefix proba --force /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig

*************************DETAILS***********************************
Last 10 commands run before the error (/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Logs/COMMANDS.log)
|2014-11-25 15:05:12| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/.asm.tigr /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.asm.tigr
|2014-11-25 15:05:12| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.contig.cnt
|2014-11-25 15:05:12| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/.contig.cnt /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.contig.cnt
|2014-11-25 15:05:12| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.contig.cvg
|2014-11-25 15:05:12| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/.contig.cvg /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.contig.cvg
|2014-11-25 15:05:12| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.afg
|2014-11-25 15:05:12| rm /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Logs/findorfs.ok
|2014-11-25 15:05:12| unlink /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig
|2014-11-25 15:05:12| ln /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Assemble/out/proba.asm.contig /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig
|2014-11-25 15:05:13| /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/prokka  --outdir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/out/proba.prokka --prefix proba --force /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig

Last 10 lines of output (/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Logs/FINDORFS.log)
[15:05:13] Determined parallel version is 20130822
[15:05:13] Looking for &#39;prodigal&#39; - found /media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/cpp/Linux-x86_64/prokka/bin/../binaries/linux/prodigal
[15:05:13] Determined prodigal version is 2.60
[15:05:13] Looking for &#39;sed&#39; - found /bin/sed
[15:05:13] Looking for &#39;signalp&#39; - found /usr/local/bin/signalp
[15:05:13] Determined signalp version is 4.1
[15:05:13] Looking for &#39;tbl2asn&#39; - found /usr/local/bin/tbl2asn
[15:05:13] Determined tbl2asn version is 23.9
[15:05:13] Using genetic code table 11.
[15:05:13] &#39;/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/FindORFS/in/proba.asm.contig&#39; is not a readable non-empty FASTA file

Please veryify input data and restart MetAMOS. If the problem persists please contact the MetAMOS development team.
*************************ERROR***********************************
*****************************************************************
rm: cannot remove ‘/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555300/Logs/findorfs.ok’: No such file or directory
Oops, MetAMOS finished with errors! see text in red above for details.
Traceback (most recent call last):
  File &quot;./runPipeline&quot;, line 984, in &lt;module&gt;
    verbose = 1)
  File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 2965, in pipeline_run
    raise job_errors
RethrownJobError: 



    Exception #1
      &#39;exceptions.TypeError(exceptions must be old-style classes or derived from BaseException, not NoneType)&#39; raised in ...
       Task = def validate.Validate(...):
       Job  = [[abyss.91.faa, idba-ud.91.faa, masurca.91.faa, sga.91.faa, soapdenovo2.91.faa, spades.91.faa, velvet.91.faa] -&gt; [validate.ok]]

    Traceback (most recent call last):
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 625, in run_pooled_job_without_exceptions
        return_value =  job_wrapper(param, user_defined_work_func, register_cleanup, touch_files_only)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 491, in job_wrapper_io_files
        ret_val = user_defined_work_func(*param)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/validate.py&quot;, line 690, in Validate
        FindORFS(&quot;%s/Assemble/out/%s.contig.cvg&quot;%(_settings.rundir, _settings.PREFIX), &quot;%s/Assemble/out/%s.faa&quot;%(_settings.rundir, _settings.PREFIX))
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/findorfs.py&quot;, line 380, in FindORFS
        findFastaORFs(_orf, &quot;%s/FindORFS/in/%s.asm.contig&quot;%(_settings.rundir, _settings.PREFIX), &quot;%s.ctg.fna&quot;%(_settings.PREFIX), &quot;%s.ctg.faa&quot;%(_settings.PREFIX), &quot;%s.ctg.gene.cvg&quot;%(_settings.PREFIX), &quot;%s.ctg.gene.map&quot;%(_settings.PREFIX), _min_ctg_len, _min_ctg_cvg)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/findorfs.py&quot;, line 337, in findFastaORFs
        run_process(_settings, &quot;%s/prokka %s --outdir %s/FindORFS/out/%s.prokka --prefix %s --force %s&quot;%(_settings.PROKKA,&quot;--fast&quot; if _run_fast else &quot;&quot;,  _settings.rundir, _settings.PREFIX, _settings.PREFIX, contigs), &quot;FindORFS&quot;)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/utils.py&quot;, line 1145, in run_process
        raise
    TypeError: exceptions must be old-style classes or derived from BaseException, not NoneType
</pre></div>


<h3>Removed <span class="caps">SRR1555298</span> and <span class="caps">SRR1555300</span>&nbsp;directories</h3>
<p>new initPipeline and run for <span class="caps">SRR1555298</span>
  nolson@Selenium:/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3$ ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555298_1.fastq  -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555298_2.fastq -i 500:1000 -W imetAMOS -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555298</span></p>
<div class="highlight"><pre>Project dir /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555298 successfully created!
Use runPipeline.py to start Pipeline

*** metAMOS running command: rm -f /media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_structure/SRR1555298/Preprocess/out/tmp.fasta

Oops, MetAMOS finished with errors! see text in red above for details.
Traceback (most recent call last):
  File &quot;./runPipeline&quot;, line 984, in &lt;module&gt;
    verbose = 1)
  File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 2965, in pipeline_run
    raise job_errors
RethrownJobError: 



    Exception #1
      &#39;exceptions.ValueError(invalid literal for int() with base 10: &#39;42\n42&#39;)&#39; raised in ...
       Task = def assemble.Assemble(...):
       Job  = [idba-ud.67.run -&gt; idba-ud.67.asm.contig]

    Traceback (most recent call last):
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 625, in run_pooled_job_without_exceptions
        return_value =  job_wrapper(param, user_defined_work_func, register_cleanup, touch_files_only)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/Utilities/ruffus/task.py&quot;, line 491, in job_wrapper_io_files
        ret_val = user_defined_work_func(*param)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/assemble.py&quot;, line 537, in Assemble
        generic.execute(STEP_NAMES.ASSEMBLE, asmName.lower(), _settings)
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 493, in execute
        return dispatch.execute()
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 342, in execute
        (havePE, haveMP, offset, input, pelist, mplist, techParams) = self.getLibInput()
      File &quot;/media/nolson/second/current_projects/micro_rm_dev/utilities/metAMOS-1.5rc3/src/generic.py&quot;, line 251, in getLibInput
        if (int(libReadLen) &lt; self.readLen):
    ValueError: invalid literal for int() with base 10: &#39;42\n42&#39;
</pre></div>


<p>Errors could be due to memory usage by pathoreport
Restarted after with pathoreport not running, same error, trying another dataset.
Potential issue with inaccurate estimate in insert size or bug in code&nbsp;&#8230;</p>
<h2>Running iMetAMOS on MiSeq data for <span class="caps">RM8375</span></h2>
<p>Ran <span class="caps">SRR1555302</span>
  ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555302_1.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555302_2.fastq -i 500:700 -W imetAMOS -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555302</span>
  ./runPipeline -v -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555302</span>/</p>
<p>Started run <span class="caps">SRR1555304</span>
  ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555304_1.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555304_2.fastq -i 500:700 -W imetAMOS -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555304</span>
  ./runPipeline -v -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555304</span>/</p>
<p>Started run <span class="caps">SRR1555306</span> 11/27/2014 10 <span class="caps">AM</span>
  ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555306_1.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555306_2.fastq -i 500:700 -W imetAMOS -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555306</span>
  ./runPipeline -v -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555306</span>/
used screen -L to log activity and ctl+a+d to detach screen
restarted pipeline with -f Scaffold and -f&nbsp;FunctionalAnnotation</p>
<p>Started run <span class="caps">SRR1555308</span> 11/27/2014 11 <span class="caps">PM</span>
  ./initPipeline -q -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555308_1.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555308_2.fastq -i 500:700 -W imetAMOS -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555308</span>
  ./runPipeline -v -d ../../analysis/bioinf/genome_structure/<span class="caps">SRR1555308</span>/</p>
            </div>
            <!-- /.entry-content -->
        </article>
    </section>

        </div>
        <div class="col-sm-3" id="sidebar">
            <aside>

<section class="well well-sm">
    <ul class="list-group list-group-flush">
            <li class="list-group-item"><h4><i class="fa fa-home fa-lg"></i><span class="icon-label">Social</span></h4>
              <ul class="list-group" id="social">
                <li class="list-group-item"><a href="#"><i class="fa fa-you-can-add-links-in-your-config-file-square fa-lg"></i> You can add links in your config file</a></li>
                <li class="list-group-item"><a href="#"><i class="fa fa-another-social-link-square fa-lg"></i> Another social link</a></li>
              </ul>
            </li>



            <li class="list-group-item"><a href="http://nate-d-olson.github.io/lab_notebok/"><h4><i class="fa fa-tags fa-lg"></i><span class="icon-label">Tags</span></h4></a>
                <ul class="list-group " id="tags">
                    <li class="list-group-item tag-1">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/docker.html">
                            Docker
                        </a>
                    </li>
                    <li class="list-group-item tag-1">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/consensus-base.html">
                            consensus base
                        </a>
                    </li>
                    <li class="list-group-item tag-2">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/ipynb.html">
                            ipynb
                        </a>
                    </li>
                    <li class="list-group-item tag-3">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/ipython-notebook.html">
                            ipython-notebook
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/pathoscope.html">
                            Pathoscope
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/sumatra.html">
                            sumatra
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/lab-notebook.html">
                            lab notebook
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/pilon.html">
                            Pilon
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/virutalenv.html">
                            virutalenv
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/imetamos.html">
                            iMetAMOS
                        </a>
                    </li>
                </ul>
            </li>
    <li class="list-group-item"><h4><i class="fa fa-external-link-square fa-lg"></i><span class="icon-label">Links</span></h4>
      <ul class="list-group" id="links">
        <li class="list-group-item">
            <a href="https://github.com/nate-d-olson" target="_blank">
                GitHub
            </a>
        </li>
        <li class="list-group-item">
            <a href="http://python.org/" target="_blank">
                Python.org
            </a>
        </li>
        <li class="list-group-item">
            <a href="http://jinja.pocoo.org/" target="_blank">
                Jinja2
            </a>
        </li>
        <li class="list-group-item">
            <a href="#" target="_blank">
                You can modify those links in your config file
            </a>
        </li>
      </ul>
    </li>
    </ul>
</section>
            </aside>
        </div>
    </div>
</div>
<footer>
   <div class="container">
      <hr>
      <div class="row">
         <div class="col-xs-10">&copy; 2015 Nate Olson
            &middot; Powered by <a href="https://github.com/DandyDev/pelican-bootstrap3" target="_blank">pelican-bootstrap3</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>         </div>
         <div class="col-xs-2"><p class="pull-right"><i class="fa fa-arrow-up"></i> <a href="#">Back to top</a></p></div>
      </div>
   </div>
</footer>
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/respond.min.js"></script>


</body>
</html>