<!DOCTYPE html>
<html lang="en"
>
<head>
    <title>Pathoscope install and pipeline development notes - Research Notebook</title>
    <!-- Using the latest rendering mode for IE -->
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">



<link rel="canonical" href="http://nate-d-olson.github.io/lab_notebok/pathoscope pipeline.html">

        <meta name="author" content="Nate Olson" />
        <meta name="keywords" content="Pathoscope,Docker" />
        <meta name="description" content="Installing pathscope, with an attempt to use docker. Unable to run with MySQL in docker. Ended up installing it locally on desktop with Ubuntu 14.04." />

        <meta property="og:site_name" content="Research Notebook" />
        <meta property="og:type" content="article"/>
        <meta property="og:title" content="Pathoscope install and pipeline development notes"/>
        <meta property="og:url" content="http://nate-d-olson.github.io/lab_notebok/pathoscope pipeline.html"/>
        <meta property="og:description" content="Installing pathscope, with an attempt to use docker. Unable to run with MySQL in docker. Ended up installing it locally on desktop with Ubuntu 14.04."/>
        <meta property="article:published_time" content="2014-11-18" />
            <meta property="article:section" content="Micro RM" />
            <meta property="article:tag" content="Pathoscope" />
            <meta property="article:tag" content="Docker" />
            <meta property="article:author" content="Nate Olson" />


    <!-- Bootstrap -->
        <link rel="stylesheet" href="http://nate-d-olson.github.io/lab_notebok/theme/css/bootstrap.min.css" type="text/css"/>
    <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/font-awesome.min.css" rel="stylesheet">

    <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/pygments/native.css" rel="stylesheet">
        <link href="http://nate-d-olson.github.io/lab_notebok/theme/css/typogrify.css" rel="stylesheet">
    <link rel="stylesheet" href="http://nate-d-olson.github.io/lab_notebok/theme/css/style.css" type="text/css"/>

        <link href="http://nate-d-olson.github.io/lab_notebok/feeds/all.atom.xml" type="application/atom+xml" rel="alternate"
              title="Research Notebook ATOM Feed"/>

</head>
<body>

<div class="navbar navbar-default navbar-fixed-top" role="navigation">
	<div class="container">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-ex1-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a href="http://nate-d-olson.github.io/lab_notebok/" class="navbar-brand">
Research Notebook            </a>
        </div>
        <div class="collapse navbar-collapse navbar-ex1-collapse">
            <ul class="nav navbar-nav">
                        <li >
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/ipynb.html">Ipynb</a>
                        </li>
                        <li class="active">
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/micro-rm.html">Micro rm</a>
                        </li>
                        <li >
                            <a href="http://nate-d-olson.github.io/lab_notebok/category/productivity.html">Productivity</a>
                        </li>
            </ul>
            <ul class="nav navbar-nav navbar-right">
              <li><a href="http://nate-d-olson.github.io/lab_notebok/archives.html"><i class="fa fa-th-list"></i><span class="icon-label">Archives</span></a></li>
            </ul>
        </div>
        <!-- /.navbar-collapse -->
    </div>
</div> <!-- /.navbar -->
<!-- Banner -->
<!-- End Banner -->
<div class="container">
    <div class="row">
        <div class="col-sm-9">

    <section id="content">
        <article>
            <header class="page-header">
                <h1>
                    <a href="http://nate-d-olson.github.io/lab_notebok/pathoscope pipeline.html"
                       rel="bookmark"
                       title="Permalink to Pathoscope install and pipeline development notes">
                        Pathoscope install and pipeline development&nbsp;notes
                    </a>
                </h1>
            </header>
            <div class="entry-content">
                <div class="panel">
                    <div class="panel-body">
<footer class="post-info">
    <span class="label label-default">Date</span>
    <span class="published">
        <i class="fa fa-calendar"></i><time datetime="2014-11-18T00:00:00+01:00"> Tue 18 November 2014</time>
    </span>



<span class="label label-default">Tags</span>
	<a href="http://nate-d-olson.github.io/lab_notebok/tag/pathoscope.html">Pathoscope</a>
        /
	<a href="http://nate-d-olson.github.io/lab_notebok/tag/docker.html">Docker</a>
    
</footer><!-- /.post-info -->                    </div>
                </div>
                <h2>Pathoscope install and pipeline development&nbsp;notes</h2>
<p>May want to revist and organize
need to install PySQLdb - python packages&nbsp;http://sourceforge.net/projects/mysql-python/</p>
<p>Approach
1. docker run -v <code>pwd</code>/micro_rm_dev -t -i micro_rm /bin/bash 
1. install MySQL in docker - command line
    1. apt-get install -y mysql-server libmysqlclient-dev python-dev python-setuptools
    2. when prompted during install entered the password for root as&nbsp;pathoscope</p>
<div class="highlight"><pre>1. STILL NEED TO DO! - add command to dockerfile
</pre></div>


<ol>
<li>set username and password - username root, password&nbsp;pathoscope</li>
<li>
<p>install pysqldb -&nbsp;http://sourceforge.net/projects/mysql-python/</p>
<p>git clone https://github.com/farcepest/MySQLdb1.git
cd MySQLdb1
python setup.py build
python setup.py&nbsp;install</p>
</li>
</ol>
<p>Issue with running mysql in a Docker container
*Followed tutorial to setup database on desktop - will need to revisit docker install later&nbsp;ftp://ftp.heanet.ie/disk1/sourceforge/p/pa/pathoscope/archive/pathoscope2.0_tutorial.pdf</p>
<p>Downloaded pathodb.sql.gz from ftp://oligomer.bumc.&nbsp;bu.edu/data/pathodb.sql.gz</p>
<p>ran the following to setup the database
mysql -u root -p
<Enter root password>
create <span class="caps">DATABASE</span> pathodb;
create user pathoscope;
grant all privileges on pathodb.* to pathoscope@”localhost” identified by 
‘johnsonlab’;
flush&nbsp;privileges;</p>
<p>mysql -u pathoscope -p pathodb &lt; pathodb.sql
<Enter the following password when asked>
johnsonlab</p>
<ol>
<li>
<p>run pathoscope lib with database info
getting into docker container using&nbsp;bash</p>
</li>
<li>
<p>getting into docker container
sudo docker run -t -i -v /media/nolson/second/current_projects/micro_rm_dev:/micro_rm  micro_rm:mysql&nbsp;/bin/bash</p>
</li>
</ol>
<p>setting up database 
when running mysql -u -p
<span class="caps">ERROR</span> 2002 (<span class="caps">HY000</span>): Can&#8217;t connect to local MySQL server through socket &#8216;/var/run/mysqld/mysqld.sock&#8217;&nbsp;(2)</p>
<p>downloaded pathodb.sql and nt_ti.fa from pathoscope&nbsp;website</p>
<h1>setting up&nbsp;pathodb</h1>
<p>gunzip -dc nt_it.gz&nbsp;&gt;nt_ti.fa</p>
<p>python ../pathoscope2/pathoscope/pathoscope.py <span class="caps">LIB</span> -genomeFile nt_ti.fa &#8212;online -dbhost localhost -dbuser pathoscope -dbpasswd johnsonlab -outPrefix&nbsp;micro_rm_patho_db</p>
<p>python pathoscope.py <span class="caps">MAP</span> -U fastq -targetRefFiles nt.fasta  -outDir results -outAlign <input_name>.sam -expTag LT2_micro_rm
used -U for single end and -1, -2 for paired&nbsp;end</p>
<p>Test run on individual file
../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">MAP</span> -1 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_1.fastq -2 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_2.fastq -targetRefFiles ../../../utilities/patho_utils/nt_ti.fa -outDir <span class="caps">SRR155296</span> -outAlign <span class="caps">SRR155296</span>.sam -expTag <span class="caps">RM8375</span>
- started around 8:30 <span class="caps">AM</span> 11/20/2014
- indexing finished around 1 <span class="caps">PM</span> 11/21/2014
Error with run, output directory needs to exist
rerun after mkdir <span class="caps">SRR1555296</span>
../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">MAP</span> -1 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_1.fastq -2 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_2.fastq -targetRefFiles ../../../utilities/patho_utils/micro_rm_patho_db_ti.fa -indexDir ../../../utilities/patho_utils/ -outDir <span class="caps">SRR1555296</span> -outAlign <span class="caps">SRR155296</span>.sam -expTag <span class="caps">RM8375</span>
-started around 1:30 <span class="caps">PM</span> 11/21/2014
-appending alignment files started at 7:<span class="caps">16PM</span></p>
<p><span class="caps">ID</span> - do not need to run if running <span class="caps">REP</span></p>
<p><span class="caps">REP</span>
need to run after pathomap finishes
../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">REP</span> -dbhost localhost -dbuser pathoscope -dbpasswd johnsonlab outDir <span class="caps">SRR1555296</span> -samfile <span class="caps">SRR1555296</span>/<span class="caps">SRR155296</span>.sam
started 6:30 <span class="caps">PM</span> on 11/23 from&nbsp;home</p>
<p>also look into using&nbsp;pathoQC</p>
<p>Patho rep still running 11/25/2014 8:20 <span class="caps">AM</span> - 13.2% of&nbsp;ram</p>
<p>Need to run patho <span class="caps">ID</span> before patho report
started at 8:50 <span class="caps">AM</span> 11/25/2014
nolson@Selenium:/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_purity/<span class="caps">SRR1555296</span>$ python ../../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">ID</span> -alignFile <span class="caps">SRR155296</span>.sam -fileType sam -outDir ./ -expTag <span class="caps">RM8375</span>-<span class="caps">ID</span></p>
<p>finished at 11:27 inlcuding updated sam&nbsp;file</p>
<p>started report as 12:40 <span class="caps">PM</span> 11/25/2014
python ../../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">REP</span> -dbhost localhost -dbuser pathoscope -dbpasswd johnsonlab -outDir ./ -samfile&nbsp;updated_SRR155296.sam</p>
<p>Used up all ram and froze computer after generating updated tsv, need to look into what is going on&nbsp;&#8230;.</p>
<p>Installing pathoqc
python pathoqc.py -1 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_1.fastq -2 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_2.fastq -s Illumina -p 8 -o ../../../analysis/bioinf/genome_structure/<span class="caps">SRR1555296</span></p>
<p>errors with pp install
able to run from within the pathoqc directory&nbsp;&#8230;.</p>
<p>Using pathoscope to characterize sequence data from reference material&nbsp;charaterization</p>
<h1>link to&nbsp;pathoscope</h1>
<p><span class="caps">PATHO</span>=../../src/pathoscope2/pathoscope/pathoscope.py</p>
<p>download the refseq&nbsp;database </p>
<p>documentation on databases
ftp://ftp.ncbi.nlm.nih.gov/blast/db/<span class="caps">README</span></p>
<p>Downloaded 11/03/2014
ftp://ftp.ncbi.nih.gov/blast/db/<span class="caps">FASTA</span>/nt.gz 20.5 Gb 
moved to utilities directory
md5sum ../../utilities/nt.gz 
670e984e61faf2aa11a074316ab85fa2&nbsp;../../utilities/nt.gz</p>
<p>Inititaing libary
python $<span class="caps">PATHO</span> <span class="caps">LIB</span> -genomeFile ../../utilities/nt.gz -outPrefix&nbsp;../../utilities/patho_nt_db</p>
<p>Mapping S0h-1 dataset
python $<span class="caps">PATHO</span> <span class="caps">MAP</span> -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/S0h-1_S1_L001_R1_001.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/S0h-1_S1_L001_R2_001.fastq -targetRefFiles ../../utilities/nt.fasta -indexDir&nbsp;../../utilities</p>
<p>Started around 10:30 am - indexing 11/4, finished ~ 4 pm&nbsp;11/6</p>
<p>Started mapping run ~4p 11/6, needs to map to the 20 split&nbsp;files</p>
<p>for 11/7 - telework
pathomap finished running - 200 + Gb of sam&nbsp;files</p>
<p>running pathoid
nohup python $<span class="caps">PATHO</span> <span class="caps">ID</span> &#8212;noUpdatedAlignFile -alignFile outalign.sam <span class="amp">&amp;</span>
started around 11:45 am&nbsp;11/7</p>
<p>running path report
nohup python ../../src/pathoscope2/pathoscope/pathoscope.py <span class="caps">REP</span> -samfile outalign.sam <span class="amp">&amp;</span>
started 9 pm&nbsp;11/7</p>
<p>Still running 11/12 - need to generate MySql database
    11/13 still running 17% of memory being used
    11/14 still running 18.2% memory &#8230;
Still Running - 1:45 11/10
Pathoscope tutorial&nbsp;http://www.microbiomejournal.com/content/supplementary/2049-2618-2-33-s5.pdf</p>
<p>Setting up mysql db
username root with not password
mysql -u root -p
<Enter root password>
create <span class="caps">DATABASE</span> pathodb;
create user pathoscope;
grant all privileges on pathodb.* to pathoscope@”localhost” identified by 
‘johnsonlab’;
flush&nbsp;privileges;</p>
<p>waiting for databases to&nbsp;download</p>
<ul>
<li>Followed pathoscope tutorial to setup PathoDB
ftp://ftp.heanet.ie/disk1/sourceforge/p/pa/pathoscope/archive/pathoscope2.0_tutorial.pdf<ul>
<li>Authors provide a curated nt GenBank database created October&nbsp;2013</li>
<li>Downloaded pathodb.sql.gz from ftp://oligomer.bumc.
bu.edu/data/pathodb.sql.gz on&nbsp;11/19/2014</li>
<li>Downloaded nt sequence data from ftp://oligomer.
bumc.bu.edu/data/nt_ti.fa.gz on&nbsp;11/19/2014</li>
</ul>
</li>
<li>Using bowtie 2 version 2.2.3 for mapping, version installed as part of&nbsp;biolinux8</li>
</ul>
<p>downloaded pathodb.sql and nt_ti.fa from pathoscope&nbsp;website</p>
<h1>setting up&nbsp;pathodb</h1>
<p>gunzip -dc nt_it.gz&nbsp;&gt;nt_ti.fa</p>
<p>python ../pathoscope2/pathoscope/pathoscope.py <span class="caps">LIB</span> -genomeFile nt_ti.fa &#8212;online -dbhost localhost -dbuser pathoscope -dbpasswd johnsonlab -outPrefix&nbsp;micro_rm_patho_db</p>
<p>python pathoscope.py <span class="caps">MAP</span> -U fastq -targetRefFiles nt.fasta  -outDir results -outAlign <input_name>.sam -expTag LT2_micro_rm
used -U for single end and -1, -2 for paired&nbsp;end</p>
<p>Test run on individual file
../../../utilities/pathoscope2/pathoscope/pathoscope.py <span class="caps">MAP</span> -1 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_1.fastq -2 ../../../data/<span class="caps">RM8375</span>/MiSeq/fastq/SRR1555296_2.fastq -targetRefFiles ../../../utilities/patho_utils/micro_rm_patho_db_ti.fa -outDir <span class="caps">SRR155296</span> -outAlign <span class="caps">SRR155296</span>.sam -expTag <span class="caps">RM8375</span></p>
<p><span class="caps">ID</span> - if running patho report do not need to run <span class="caps">ID</span></p>
<p><span class="caps">REP</span></p>
<p>also look into using pathoQC&nbsp;http://sourceforge.net/projects/pathoscope/files/pathoqc_v0.1.1.tar.gz/download</p>
<ul>
<li>Started test run at 10:00 <span class="caps">PM</span>&nbsp;12/4/2014</li>
<li>
<p>tomorrow </p>
<ul>
<li>check&nbsp;run</li>
<li>optimize - make sure using all 8 cores, or parallelize runs<ul>
<li>look into best way to optimize, paralleize multiple samples using a single core for each or running <span class="caps">ID</span> after map in&nbsp;parallel</li>
</ul>
</li>
<li>issue with defining the number of threads - change manually for&nbsp;now</li>
<li>update readme with code for running the&nbsp;pipeline</li>
<li>error with cleanup rm: cannot remove ‘/media/nolson/second/current_projects/micro_rm_dev/analysis/bioinf/genome_purity/<span class="caps">SRR1393721</span>/<span class="caps">SRR1393721</span>-utilities/patho_utils/micro_rm_patho_db_ti.*sam’: No such file or&nbsp;directory</li>
<li>update code to that steps will not&nbsp;rerun</li>
</ul>
</li>
<li>
<p>restarted text 8:30 <span class="caps">AM</span>&nbsp;12/5/2015</p>
</li>
</ul>
<p>Using pathoscope to characterize sequence data from reference material&nbsp;charaterization</p>
<h1>link to&nbsp;pathoscope</h1>
<p><span class="caps">PATHO</span>=../../src/pathoscope2/pathoscope/pathoscope.py</p>
<p>download the refseq&nbsp;database </p>
<p>documentation on databases
ftp://ftp.ncbi.nlm.nih.gov/blast/db/<span class="caps">README</span></p>
<p>Downloaded 11/03/2014
ftp://ftp.ncbi.nih.gov/blast/db/<span class="caps">FASTA</span>/nt.gz 20.5 Gb 
moved to utilities directory
md5sum ../../utilities/nt.gz 
670e984e61faf2aa11a074316ab85fa2&nbsp;../../utilities/nt.gz</p>
<p>Inititaing libary
python $<span class="caps">PATHO</span> <span class="caps">LIB</span> -genomeFile ../../utilities/nt.gz -outPrefix&nbsp;../../utilities/patho_nt_db</p>
<p>Mapping S0h-1 dataset
python $<span class="caps">PATHO</span> <span class="caps">MAP</span> -1 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/S0h-1_S1_L001_R1_001.fastq -2 ../../data/<span class="caps">RM8375</span>/MiSeq/fastq/S0h-1_S1_L001_R2_001.fastq -targetRefFiles ../../utilities/nt.fasta -indexDir&nbsp;../../utilities</p>
<p>Started around 10:30 am - indexing 11/4, finished ~ 4 pm&nbsp;11/6</p>
<p>Started mapping run ~4p 11/6, needs to map to the 20 split&nbsp;files</p>
<p>for 11/7 - telework
pathomap finished running - 200 + Gb of sam&nbsp;files</p>
<p>running pathoid
nohup python $<span class="caps">PATHO</span> <span class="caps">ID</span> &#8212;noUpdatedAlignFile -alignFile outalign.sam <span class="amp">&amp;</span>
started around 11:45 am&nbsp;11/7</p>
<p>running path report
nohup python ../../src/pathoscope2/pathoscope/pathoscope.py <span class="caps">REP</span> -samfile outalign.sam <span class="amp">&amp;</span>
started 9 pm&nbsp;11/7</p>
<p>Still running 11/12 - need to generate MySql database
    11/13 still running 17% of memory being used
    11/14 still running 18.2% memory &#8230;
Still Running - 1:45 11/10
Pathoscope tutorial&nbsp;http://www.microbiomejournal.com/content/supplementary/2049-2618-2-33-s5.pdf</p>
<p>Setting up mysql db
username root with not password
mysql -u root -p
<Enter root password>
create <span class="caps">DATABASE</span> pathodb;
create user pathoscope;
grant all privileges on pathodb.* to pathoscope@”localhost” identified by 
‘johnsonlab’;
flush&nbsp;privileges;</p>
<p>waiting for databases to&nbsp;download</p>
            </div>
            <!-- /.entry-content -->
        </article>
    </section>

        </div>
        <div class="col-sm-3" id="sidebar">
            <aside>

<section class="well well-sm">
    <ul class="list-group list-group-flush">
            <li class="list-group-item"><h4><i class="fa fa-home fa-lg"></i><span class="icon-label">Social</span></h4>
              <ul class="list-group" id="social">
                <li class="list-group-item"><a href="#"><i class="fa fa-you-can-add-links-in-your-config-file-square fa-lg"></i> You can add links in your config file</a></li>
                <li class="list-group-item"><a href="#"><i class="fa fa-another-social-link-square fa-lg"></i> Another social link</a></li>
              </ul>
            </li>



            <li class="list-group-item"><a href="http://nate-d-olson.github.io/lab_notebok/"><h4><i class="fa fa-tags fa-lg"></i><span class="icon-label">Tags</span></h4></a>
                <ul class="list-group " id="tags">
                    <li class="list-group-item tag-1">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/docker.html">
                            Docker
                        </a>
                    </li>
                    <li class="list-group-item tag-2">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/ipynb.html">
                            ipynb
                        </a>
                    </li>
                    <li class="list-group-item tag-2">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/consensus-base.html">
                            consensus base
                        </a>
                    </li>
                    <li class="list-group-item tag-3">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/ipython-notebook.html">
                            ipython-notebook
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/virutalenv.html">
                            virutalenv
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/pilon.html">
                            Pilon
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/pathoscope.html">
                            Pathoscope
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/sumatra.html">
                            sumatra
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/imetamos.html">
                            iMetAMOS
                        </a>
                    </li>
                    <li class="list-group-item tag-4">
                        <a href="http://nate-d-olson.github.io/lab_notebok/tag/lab-notebook.html">
                            lab notebook
                        </a>
                    </li>
                </ul>
            </li>
    <li class="list-group-item"><h4><i class="fa fa-external-link-square fa-lg"></i><span class="icon-label">Links</span></h4>
      <ul class="list-group" id="links">
        <li class="list-group-item">
            <a href="https://github.com/nate-d-olson" target="_blank">
                GitHub
            </a>
        </li>
        <li class="list-group-item">
            <a href="http://python.org/" target="_blank">
                Python.org
            </a>
        </li>
        <li class="list-group-item">
            <a href="http://jinja.pocoo.org/" target="_blank">
                Jinja2
            </a>
        </li>
        <li class="list-group-item">
            <a href="#" target="_blank">
                You can modify those links in your config file
            </a>
        </li>
      </ul>
    </li>
    </ul>
</section>
            </aside>
        </div>
    </div>
</div>
<footer>
   <div class="container">
      <hr>
      <div class="row">
         <div class="col-xs-10">&copy; 2015 Nate Olson
            &middot; Powered by <a href="https://github.com/DandyDev/pelican-bootstrap3" target="_blank">pelican-bootstrap3</a>,
            <a href="http://docs.getpelican.com/" target="_blank">Pelican</a>,
            <a href="http://getbootstrap.com" target="_blank">Bootstrap</a>         </div>
         <div class="col-xs-2"><p class="pull-right"><i class="fa fa-arrow-up"></i> <a href="#">Back to top</a></p></div>
      </div>
   </div>
</footer>
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/jquery.min.js"></script>

<!-- Include all compiled plugins (below), or include individual files as needed -->
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/bootstrap.min.js"></script>

<!-- Enable responsive features in IE8 with Respond.js (https://github.com/scottjehl/Respond) -->
<script src="http://nate-d-olson.github.io/lab_notebok/theme/js/respond.min.js"></script>


</body>
</html>